src/target/at32f43x.c

```diff
@@ -364,3 +388,150 @@ static bool at32f43_mass_erase(target_s *target)
 		return at32f43_mass_erase_bank(target, AT32F43x_FLASH_BANK2_REG_OFFSET, &timeout);
 	return true;
 }
+
+static bool at32f43_option_erase(target_s *target)
+{
+	/* bank_reg_offset is 0, option bytes belong to first bank */
+	at32f43_flash_clear_eop(target, 0);
+
+	/* Wipe User System Data */
+	target_mem32_write32(target, AT32F43x_FLASH_CTRL, AT32F43x_FLASH_CTRL_USDERS | AT32F43x_FLASH_CTRL_USDULKS);
+	target_mem32_write32(target, AT32F43x_FLASH_CTRL,
+		AT32F43x_FLASH_CTRL_USDERS | AT32F43x_FLASH_CTRL_USDULKS | AT32F43x_FLASH_CTRL_ERSTR);
+
+	return at32f43_flash_busy_wait(target, 0, NULL);
+}
+
+static bool at32f43_option_write_erased(target_s *const target, const size_t offset, const uint16_t value)
+{
+	if (value == 0xffffU)
+		return true;
+
+	at32f43_flash_clear_eop(target, 0);
+
+	/* Enable writing User System Data */
+	target_mem32_write32(target, AT32F43x_FLASH_CTRL, AT32F43x_FLASH_CTRL_USDPRGM | AT32F43x_FLASH_CTRL_USDULKS);
+
+	const uint32_t addr = AT32F43x_USD_BASE + (offset * 2U);
+	target_mem32_write16(target, addr, value);
+
+	const bool result = at32f43_flash_busy_wait(target, 0, NULL);
+	if (result || offset != 0U)
+		return result;
+
+	/* For error on offset 0, that is RDP byte, signal back the failure to erase RDP (?) */
+	const uint8_t status =
+		target_mem32_read32(target, AT32F43x_FLASH_STS) & (AT32F43x_FLASH_STS_PRGMERR | AT32F43x_FLASH_STS_EPPERR);
+	return status == AT32F43x_FLASH_STS_PRGMERR;
+}
+
+#define STM32F1_OB_COUNT     8U
+#define AT32F43x_2K_OB_COUNT 256U
+#define AT32F43x_4K_OB_COUNT 2048U
```

Comment:
We're glad to see these defined like this, just they should up at the top with the other defines please.